app.js

```diff
@@ -57,12 +57,16 @@ app.use('/oauthSignin', (req, res) => {
         elId: req.query.elementId
     };
     req.session.state = state;
-    console.log("In /oauthSignin middleware");
     return passport.authenticate('onshape', { state: uuid.v4(state) })(req, res);
-}, (req, res) => {     console.log("Redirecting to Onshape for authentication");/* redirected to Onshape for authentication */ });
+}, (req, res) => { /* redirected to Onshape for authentication */ });
 
 app.use('/oauthRedirect', passport.authenticate('onshape', { failureRedirect: '/grantDenied' }), (req, res) => {
-    res.redirect(`/?documentId=${req.session.state.docId}&workspaceId=${req.session.state.workId}&elementId=${req.session.state.elId}`);
+    if (req.session.state) {
+        res.redirect(`/?documentId=${req.session.state.docId}&workspaceId=${req.session.state.workId}&elementId=${req.session.state.elId}`);
+    }
+    else {
+        res.sendFile(path.join(__dirname, 'public', 'html', 'cookiesDisabled.html'));
```

Comment:
Can you add a comment here as to what this path of code is? For example // display third party cookies disabled if unable to save req.session.state

Reply:
Done